### PR TITLE
add bytelength check to NewDIDFromBytes

### LINKED ISSUE #889: check byelength in method NewDIDFromBytes
<This is intended as a guide when thinking about or creating issues for go-centrifuge>

should equal common.AddressLength (DIDLength), return an err if not.

### PR DESCRIPTION
**Pull request by charleenfei**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Closes #889

### CODE REVIEW

File: documents/processor_test.go [hunk 1]
```diff
@@ -393,13 +397,14 @@ func TestDefaultProcessor_AnchorDocument(t *testing.T) {
...
 	model.On("CalculateDocumentRoot").Return(nil, errors.New("error"))
-	model.On("Author").Return(did1)
+	model.On("Author").Return(did1, nil)
 	model.On("GetSignerCollaborators", mock.Anything).Return([]identity.DID{did1, testingidentity.GenerateRandomDID()}, nil)
 	tm := time.Now()
 	model.On("Timestamp").Return(tm, nil)
 	model.sigs = append(model.sigs, sig)
 	srv = &testingcommons.MockIdentityService{}
-	srv.On("ValidateSignature", identity.NewDIDFromBytes(did), sig.PublicKey, sig.Signature, sr, tm).Return(nil).Once()
+	cid, _ := identity.NewDIDFromBytes(did)
```
Comment (vimukthi-git) [inline documents/processor_test.go]: assert no error

File: documents/processor_test.go [hunk 2]
```diff
@@ -486,13 +492,14 @@ func TestDefaultProcessor_SendDocument(t *testing.T) {
...
 	model.On("CalculateDocumentRoot").Return(utils.RandomSlice(32), nil)
-	model.On("Author").Return(did1)
+	model.On("Author").Return(did1, nil)
 	model.On("GetSignerCollaborators", mock.Anything).Return([]identity.DID{did1, testingidentity.GenerateRandomDID()}, nil)
 	tm := time.Now()
 	model.On("Timestamp").Return(tm, nil)
 	model.sigs = append(model.sigs, sig)
 	srv = &testingcommons.MockIdentityService{}
-	srv.On("ValidateSignature", identity.NewDIDFromBytes(didb), sig.PublicKey, sig.Signature, sr, tm).Return(nil).Once()
+	cid, _ := identity.NewDIDFromBytes(didb)
```
Comment (vimukthi-git) [inline documents/processor_test.go]: same here

File: documents/validator_test.go [hunk 3]
```diff
@@ -439,16 +441,18 @@ func TestValidator_signatureValidator(t *testing.T) {
...
 	model.On("CalculateSigningRoot").Return(sr, nil).Once()
 	model.On("Signatures").Return().Once()
-	model.On("Author").Return(did)
+	model.On("Author").Return(did, nil)
 	model.On("GetSignerCollaborators", mock.Anything).Return([]identity.DID{did, testingidentity.GenerateRandomDID()}, nil)
 	model.On("Timestamp").Return(tm, nil)
 	model.sigs = append(model.sigs, s)
 	srv = new(testingcommons.MockIdentityService)
-	srv.On("ValidateSignature", identity.NewDIDFromBytes(s.SignerId), s.PublicKey, s.Signature, sr, tm).Return(errors.New("error")).Once()
+	sid, _ := identity.NewDIDFromBytes(s.SignerId)
```
Comment (vimukthi-git) [inline documents/validator_test.go]: same here

File: p2p/receiver/handler_test.go [hunk 4]
```diff
@@ -218,7 +218,7 @@ func TestP2PService_basicChecks(t *testing.T) {
 	}
 
 	id, _ := cfg.GetIdentityID()
-	centID := identity.NewDIDFromBytes(id)
+	centID, err := identity.NewDIDFromBytes(id)
```
Comment (vimukthi-git) [inline p2p/receiver/handler_test.go]: same here

File: testingutils/identity/identity.go [hunk 5]
```diff
@@ -64,7 +65,8 @@ func CreateAccountIDWithKeys(contextTimeout time.Duration, acc *configstore.Acco
 
 func GenerateRandomDID() identity.DID {
 	r := utils.RandomSlice(identity.DIDLength)
-	return identity.NewDIDFromBytes(r)
+	did, _ := identity.NewDIDFromBytes(r)
```
Comment (vimukthi-git) [inline testingutils/identity/identity.go]: return error

Comment (charleenfei) [reply testingutils/identity/identity.go]: I think that in most cases it makes sense to return the error, but in this function I think it creates more work than helping to debug, as we are just generating the randomDID from a specified bytelength, which leaves no room for bugs resulting from user input. Returning the error here would result in changing over 100 different places we use this method.

**Pull request merged**